nsv13/code/modules/recycling/replicatorDS.dm

```diff
@@ -0,0 +1,359 @@
+		switch(what)
+			if("cake batter")
+				food = new /obj/item/reagent_containers/food/snacks/cakebatter(get_turf(src))
+			if("dough")
+				food = new /obj/item/reagent_containers/food/snacks/dough(get_turf(src))
+			if("egg box")
+				food = new /obj/item/storage/fancy/egg_box(get_turf(src))
+			if("flour")
+				food = new /obj/item/reagent_containers/food/condiment/flour(get_turf(src))
+			if("milk")
+				food = new /obj/item/reagent_containers/food/condiment/milk(get_turf(src))
+			if("enzymes")
+				food = new /obj/item/reagent_containers/food/condiment/enzyme(get_turf(src))
+			if("cheese wheel")
+				food = new /obj/item/reagent_containers/food/snacks/store/cheesewheel(get_turf(src))
+			if("meat slab","slab of meat")
+				food = new /obj/item/reagent_containers/food/snacks/meat/slab(get_turf(src))
+			if("activate iguana")
+				if(length(iguanas) > 9)
+					say("You have reached the iguana limit!")
+					return
+				iguanas += new /mob/living/simple_animal/kalo/leonard(get_turf(src))
+			if("deactivate iguana")
+				for(var/mob/M as() in iguanas)
+					iguanas -= M
+					qdel(M)
+
+	if(food)
+		var/nutriment = food.reagents.get_reagent_amount("nutriment")
+		if(Biogen.points >= nutriment && Biogen.points >= 5)
+			//time to check laser power.
+			if(prob(6-failure_grade)) //Chance to make a burned mess so the chef is still useful.
+				var/obj/item/reagent_containers/food/snacks/badrecipe/neelixcooking = new /obj/item/reagent_containers/food/snacks/badrecipe(get_turf(src))
+				neelixcooking.name = "replicator mess"
+				neelixcooking.desc = "perhaps you should invest in some higher quality parts."
+				Biogen.points -= 5
+				qdel(food) //NO FOOD FOR YOU!
+				return
+			else
+				if(temp)
+					food.name = "[temp] [food.name]"
+					switch(temp)
+						if("cold")
+							food.reagents.chem_temp = 0
+						if("hot")
+							food.reagents.chem_temp = 450
+						if("extra hot")
+							food.reagents.chem_temp = 5000
+						if("well done")
+							food.reagents.chem_temp = 2000000000000 //A nice warm Steak or a perfectly well boiled Cup of Tea
+				if(nutriment > 0)
+					Biogen.points -= nutriment
+				else
+					Biogen.points -= 5 //Default, in case the food is useless.
+				if(emagged)
+					food.reagents.add_reagent("munchyserum", nutriment)
+					food.reagents.remove_reagent("nutriment",nutriment)
+				var/currentHandIndex = user.get_held_index_of_item(food)
+				user.put_in_hand(food,currentHandIndex)
+
+		else
+			visible_message("<span_class='warning'> Insufficient fuel to create [food]. [src] requires [nutriment] U of biomatter.</span>")
+			qdel(food) //NO FOOD FOR YOU!
+
+
+/mob/living/simple_animal/kalo/leonard
+	name = "Leonard"
+	desc = "A holographic pet lizard. Say 'deactivate iguana' if you're a square."
+
+/datum/reagent/toxin/munchyserum //Tasteless alternative to lipolicide, less powerful. This has the reverse of the intended effect of a replicator and makes you hungrier.
+	name = "Metabolism Overide Toxin"
```

Comment:
Override

Reply:
Are you telling me to override something or that I'm overriding something that I shouldn't?

Reply:
Spelling

Reply:
Ah, thank you.